pkg/server/ca/manager/slot.go

```diff
@@ -86,74 +84,178 @@ func (s *SlotLoader) Load(ctx context.Context) (*Journal, map[SlotPosition]Slot,
 		return nil, nil, err
 	}
 
-	if len(entries.X509CAs) > 0 {
-		nextX509CA, err = s.tryLoadX509CASlotFromEntry(ctx, entries.X509CAs[len(entries.X509CAs)-1])
+	currentX509CA, nextX509CA, err := s.getX509CASlots(ctx, entries.X509CAs)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	currentJWTKey, nextJWTKey, err := s.getJWTKeysSlots(ctx, entries.JwtKeys)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	slots := make(map[SlotPosition]Slot)
+	if currentX509CA != nil {
+		slots[CurrentX509CASlot] = currentX509CA
+	}
+
+	if nextX509CA != nil {
+		slots[NextX509CASlot] = nextX509CA
+	}
+
+	if currentJWTKey != nil {
+		slots[CurrentJWTKeySlot] = currentJWTKey
+	}
+
+	if nextJWTKey != nil {
+		slots[NextJWTKeySlot] = nextJWTKey
+	}
+
+	return journal, slots, nil
+}
+
+// getX509CASlots returns slots based on status
+// - If all status are unknown, choose the two newest on the list
+// - Active entry is returned on current if set
+// - Newest Prepared or Old entry is returned on next
+func (s *SlotLoader) getX509CASlots(ctx context.Context, entries []*X509CAEntry) (*X509CASlot, *X509CASlot, error) {
+	var current *X509CASlot
+	var next *X509CASlot
+
+	// Search from oldest
+	for i := len(entries) - 1; i >= 0; i-- {
+		slot, err := s.tryLoadX509CASlotFromEntry(ctx, entries[i])
```

Comment:
Since this is now done in every entry, a lot of possible unusable slots will be found and logged. We should probably update `tryLoadX509CASlotFromEntry` to log more information about the entry, like the status and issuance time. Also, I'm wondering if entries with OLD status should really log a warning. Maybe debug level is more appropriate,.

Reply:
it is not really loading all, just until you found the active and the first prepared or old entry, generally you will be getting only the first 2 elements as before, except there was some force rotations envolved where user prepared more than one CA in advance (no regular case)